src/world/area_obk/obk_01/BBD1A0.c

```diff
@@ -31,8 +34,179 @@ ApiStatus func_8024007C_BBD21C(Evt* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
+typedef struct UnkObk01 {
+    /* 0x00 */ s32 unk_00;
+    /* 0x04 */ s32 unk_04;
+    /* 0x08 */ s32* unk_08;
+    /* 0x0C */ f32 unk_0C; // some angle
+    /* 0x10 */ f32 unk_10; // some y
+    /* 0x14 */ f32 unk_14;
+    /* 0x18 */ f32 unk_18;
+    /* 0x1C */ Matrix4f unk_1C;
+    /* 0x5C */ Model* models[4];
+} UnkObk01; // size = 0x6C;
+
+ApiStatus func_80240098_BBD238(Evt* script, s32 isInitialCall) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Bytecode* args = script->ptrReadPos;
+    Matrix4f sp18;
+    f32 temp_f20;
+    f32 var_f22;
+    s32 var_a0;
+    s32 temp_s0_2;
+    Model* model;
+    UnkObk01* temp_v0;
+    s32 i;
+
+    if (isInitialCall) {
+        script->functionTempPtr[1] = temp_v0 = heap_malloc(sizeof(*temp_v0));
+        temp_v0->unk_08 = (s32*) evt_get_variable(script, *args++);
+        temp_v0->unk_08[0] = (s32) temp_v0; // what is going on here?
```

Comment:
An evt variable holding a pointer to a struct isn't too uncommon

Reply:
I'm not sure what you mean. the question is why is temp-v0->unk_08 holding a pointer to itself

Reply:
the code is doing `x->y = x`

Reply:
Oh, I missed `x`/`temp_v0` was there twice. It's early in the day 😛 

I think this is probably because `unk_08` is a pointer that came from the outside environment (`evt_get_variable`), so something other than this function might need access to the allocated data. This theoretical other function won't be able to use `script->functionTempPtr[1]` after this API function is done, so instead it can dereference the evt variable

Reply:
I snee. maybe, but as far as I can tell, this is only used in this one function and barely 2 others

Reply:
Maybe this function is called multiple times from the script, ie the "other function" is just another call to this one